nilearn/datasets/func.py

```diff
@@ -194,11 +194,13 @@ def fetch_haxby(
                 {"uncompress": True},
             )
         ]
-        readme = fetch_files(
-            data_dir, stimuli_files, resume=resume, verbose=verbose
-        )[0]
+        readme = Path(
+            fetch_files(
+                data_dir, stimuli_files, resume=resume, verbose=verbose
+            )[0]
+        )
         kwargs["stimuli"] = tree(
-            os.path.dirname(readme), pattern="*.jpg", dictionary=True
+            readme.parent, pattern="*.jpg", dictionary=True
```

Comment:
```suggestion
            Path(readme).parent, pattern="*.jpg", dictionary=True
```

Can we not just do this here?

And keep the older above:

```python
readme = fetch_files(
            data_dir, stimuli_files, resume=resume, verbose=verbose
        )[0]
```

Feels easier to read.